packages/grafana-data/src/types/data.ts

```diff
@@ -33,6 +39,12 @@ export interface QueryResultMeta {
   /** Currently used to show results in Explore only in preferred visualisation option */
   preferredVisualisationType?: PreferredVisualisationType;
 
+  /**
+   * Optionally identify which topic the frame should be assigned to.
+   * A value specified in the response will override what the request asked for.
+   */
+  dataTopic?: DataTopic;
```

Comment:
@aocenas -- you will ask why not use the "preferredVisualisationType"  -- my understanding of this property when used in a dashboard context is that it would let the data source say which *panel* type would be a good choice.  Exemplars and annotations should exist in the same panel the series graph/table/logs etc

Reply:
well the meaning can evolve if needed I think. I mean prefVisType does not need to map 1 to 1 with panels. Like for example if somebody wants a line together with bars in one panel and create a plugin for that it may define it can show both bars and line and so could show 2 data frames with different prefVisType. I mean I am not saying this would be the best approach but this was sort of how I was thinking about it. In that case graph panel could handle multiple visTypes like Graph, Annotations (vertical lines) and Examplars (on hover click data).

Reply:
Again not sure if good idea I feel at that point in Explore it would be hard to route the data in all cases. But for example one thing we may be adding is some sort of group key to say what can be grouped how, so that if you have multiple graph results you could say if that should be 2 separate panels or rather single panel. That could be used to group exemplars and graph together so maybe it would still work.